ApplySyntax.py

```diff
@@ -274,8 +274,23 @@ def __init__(self):
         self.view = None
         self.syntaxes = []
         self.reraise_exceptions = False
+        self.seen_deprecation_warnings = {
+            'binary': False,
+            'file_name': False,
+        }
+
         on_touched_callback = self.on_touched
 
+    def print_deprecation_warning(self, keyword):
+        if self.seen_deprecation_warnings[keyword]:
+            return
+
+        self.seen_deprecation_warnings[keyword] = True
+        log(
+            "Warning: '%s' keyword is deprecated and may be removed in the future."
+            % keyword
+        )
+
```

Comment:
I like this.  I would have personally just let the warnings clutter the console and annoy them into compliance, but this is nice :).